packages/data/src/components/use-select/test/index.js

```diff
@@ -132,5 +132,91 @@ describe( 'useSelect', () => {
 			children: 'bar',
 		} );
 	} );
+	describe( 'rerenders as expected with various mapSelect return types', () => {
+		const getComponent = ( selectSpy ) => () => {
+			selectSpy.mockImplementation(
+				( select ) => select( 'testStore' ).testSelector()
+			);
+			const data = useSelect( selectSpy, [] );
+			return <div data={ data } />;
+		};
+		let selectorSpy,
+			subscribedSpy,
+			subscriberSpy,
+			selectSpy,
+			TestComponent;
+		beforeEach( () => {
+			selectorSpy = jest.fn();
+			subscriberSpy = ( subscription ) => {
+				subscribedSpy = subscription;
+			};
+			registry.registerStore( 'testStore', {
+				reducer: () => null,
+				selectors: {
+					testSelector: selectorSpy,
+				},
+			} );
+			registry.subscribe = subscriberSpy;
+			selectSpy = jest.fn();
+			TestComponent = getComponent( selectSpy );
+		} );
+		[
+			[
+				'boolean',
+				[ false, true ],
+			],
+			[
+				'number',
+				[ 10, 20 ],
+			],
+			[
+				'string',
+				[ 'bar', 'cheese' ],
+			],
+			[
+				'array',
+				[ [ 10, 20 ], [ 10, 30 ] ],
+			],
+			[
+				'object',
+				[ { foo: 'bar' }, { foo: 'cheese' } ],
+			],
+			[
+				'null',
+				[ null, undefined ],
+			],
+			[
+				'undefined',
+				[ undefined, 42 ],
+			],
+		].forEach(
```

Comment:
Why not use the `each([...]).test` approach that Jest supports natively?

Reply:
Honestly, I just didn't think to use it :).  It does look like the test could be switched to use it here but is it functionally any different?  Are there some advantages to using it here?

Reply:
Different reporting I think, and better ESLint support.

Reply:
Interesting, good points and I'll give a go at updating the tests to use the built-in test conditions api in jest.  It's also something I'm willing to do in a separate pull if this fix is needed to move #16460 along.

Reply:
implemented in e2f6c28